### PR TITLE
feat(dashboard): in-app editor unsaved changes modal plus handling esc and clicking on drawer gray overlay

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

In-App Editor Drawer:
- unsaved changes modal
- handle the ESC button and clicks on the drawer gray overlay

### Screenshots

Screenshot 2024-11-12 at 13 56 58

https://github.com/user-attachments/assets/62159e86-55e5-4e95-b334-fc23fac8258a

### CODE REVIEW

File: apps/dashboard/package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "@radix-ui/react-switch": "^1.1.1",
     "@radix-ui/react-tabs": "^1.1.1",
     "@radix-ui/react-tooltip": "^1.1.3",
+    "@radix-ui/react-visually-hidden": "^1.1.0",
```
Comment (LetItRock) [inline apps/dashboard/package.json:48]: used to hide some required components, more about it in the comments below

File: apps/dashboard/src/components/primitives/tabs.tsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ TabsTrigger.displayName = TabsPrimitive.Trigger.displayName;
 const tabsContentVariants = cva('focus-visible:outline-none', {
   variants: {
     variant: {
-      default: 'ring-offset-background focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2',
+      default: '',
```
Comment (LetItRock) [inline apps/dashboard/src/components/primitives/tabs.tsx:58]: no need to show the outline on the tabs content component

File: apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx [hunk 3]
```diff
@@ -36,24 +49,30 @@ export const EditStepSidebar = () => {
...
-          }}
-          transition={transitionSetting}
-          className={
-            'bg-background fixed inset-y-0 right-0 z-50 flex h-full w-3/4 flex-col border-l shadow-lg sm:max-w-[600px]'
-          }
-        >
-          {/* TODO: show loading indicator */}
-          {workflow && step && stepType && <StepEditor workflow={workflow} step={step} stepType={stepType} />}
-        </motion.div>
+        <SheetContentBase asChild onInteractOutside={handleCloseSidebar} onEscapeKeyDown={handleCloseSidebar}>
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx:52]: The `SheetContentBase` is the component that allows to handle the clicks on the drawer overlay and ESC button clicks. But the thing is that it requires the use of the `SheetTitle, SheetDescription` under the hood, but the current implementation doesn't use it. So I had to hide these two visually.
We might decide to refactor that in the future.

File: apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx [hunk 4]
```diff
@@ -128,75 +138,112 @@ export const InAppTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; s
...
+                form="save-step"
+                disabled={!formState.isDirty}
+              >
+                Save step
+              </Button>
+            </footer>
+          </Tabs>
+        </form>
+      </Form>
+      <AlertDialog open={blocker.state === 'blocked'}>
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx:220]: The unsaved changes modal

**Review by desiprisg — APPROVED** [review]

**Review by BiswaViraj — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx (hunk 3 above)
Comment (SokratisVidros) [reply apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx:52]: Please add a TODO in the code.

**Pull request merged**